system/HTTP/OutgoingRequest.php

```diff
@@ -66,15 +66,13 @@ private function getHostFromUri(URI $uri): string
     }
 
     /**
-     * Get the request method.
+     * Retrieves the HTTP method of the request.
      *
-     * @param bool $upper Whether to return in upper or lower case.
-     *
-     * @deprecated The $upper functionality will be removed and this will revert to its PSR-7 equivalent
+     * @return string Returns the request method.
      */
-    public function getMethod(bool $upper = false): string
+    public function getMethod(): string
     {
-        return ($upper) ? strtoupper($this->method) : strtolower($this->method);
+        return $this->method;
```

Comment:
I would feel a lot better if we could force uppercase here. Just in case, for edge cases, like some misconfigured proxy or something. According to the spec we expect uppercase, but there are no guarantees.

```suggestion
        return strtoupper($this->method);
```

Reply:
I'm not sure what your concern is.
When you say proxy, do you mean reverse proxy or forward proxy?

At least, it seems there is no server that accepts "get" method.

```console
$ curl -X get https://codeigniter.com
<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>cloudflare</center>
</body>
</html>
```
```console
$ curl -X get https://www.php.net
<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>Myra</center>
</body>
</html>
```
```console
$ curl -X get https://www.w3.org
<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>cloudflare</center>
</body>
</html>
```
```console
$ curl -X get https://www.mozilla.org
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<HTML><HEAD><META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<TITLE>ERROR: The request could not be satisfied</TITLE>
</HEAD><BODY>
<H1>400 ERROR</H1>
```
```console
$ curl -X get https://www.google.com
<!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 405 (Method Not Allowed)!!1</title>
```

Reply:
Reverse proxy. I remember I was reading about this type of issue, but it was so long ago I can't find it now. Anyway... you might be right. The spec is clear and it should be uppercase, so let's level it how it is.

Reply:
I think a good compromise would be to state that the return value is upper.